From 9e5a69ced6bf9e879a93215dc498bfb07f0fa518 Mon Sep 17 00:00:00 2001
From: Aditya Garg <gargaditya08@live.com>
Date: Tue, 18 Mar 2025 21:27:25 +0530
Subject: [PATCH] HID: magicmouse: Add MacBookPro15,1 replacement trackpad
 support

This commit adds support for third party replacement trackpad for
MacBookPro15,1 with a different PID in hid-magicmouse.

Signed-off-by: Aditya Garg <gargaditya08@live.com>
---
 drivers/hid/hid-magicmouse.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

Index: lernstick-linux/drivers/hid/hid-magicmouse.c
===================================================================
--- lernstick-linux.orig/drivers/hid/hid-magicmouse.c	2025-10-13 14:24:18.821071034 +0200
+++ lernstick-linux/drivers/hid/hid-magicmouse.c	2025-10-13 14:24:18.817070933 +0200
@@ -151,6 +151,17 @@
 #define J680_TP_RES_Y \
 	((J680_TP_MAX_Y - J680_TP_MIN_Y) / (J680_TP_DIMENSION_Y / 100))
 
+#define J680_ALT_TP_DIMENSION_X (float)16000
+#define J680_ALT_TP_MIN_X -7456
+#define J680_ALT_TP_MAX_X 7976
+#define J680_ALT_TP_RES_X \
+	((J680_ALT_TP_MAX_X - J680_ALT_TP_MIN_X) / (J680_ALT_TP_DIMENSION_X / 100))
+#define J680_ALT_TP_DIMENSION_Y (float)10000
+#define J680_ALT_TP_MIN_Y -163
+#define J680_ALT_TP_MAX_Y 9283
+#define J680_ALT_TP_RES_Y \
+	((J680_ALT_TP_MAX_Y - J680_ALT_TP_MIN_Y) / (J680_ALT_TP_DIMENSION_Y / 100))
+
 #define J213_TP_DIMENSION_X (float)13500
 #define J213_TP_MIN_X -6243
 #define J213_TP_MAX_X 6749
@@ -312,6 +323,7 @@
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J140K:
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J132:
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J680:
+		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J680_ALT:
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J213:
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J214K:
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J223:
@@ -1215,6 +1227,7 @@
 	T2_TOUCHPAD_ENTRY(J140K),
 	T2_TOUCHPAD_ENTRY(J132),
 	T2_TOUCHPAD_ENTRY(J680),
+	T2_TOUCHPAD_ENTRY(J680_ALT),
 	T2_TOUCHPAD_ENTRY(J213),
 	T2_TOUCHPAD_ENTRY(J214K),
 	T2_TOUCHPAD_ENTRY(J223),
@@ -1465,6 +1478,7 @@
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J140K:
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J132:
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J680:
+	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J680_ALT:
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J213:
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J214K:
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J223:
@@ -1497,6 +1511,7 @@
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J140K:
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J132:
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J680:
+		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J680_ALT:
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J213:
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J214K:
 		case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J223:
@@ -1572,6 +1587,7 @@
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J140K:
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J132:
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J680:
+	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J680_ALT:
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J213:
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J214K:
 	case USB_DEVICE_ID_APPLE_WELLSPRINGT2_J223:
@@ -1704,6 +1720,8 @@
 	{ HID_USB_DEVICE(USB_VENDOR_ID_APPLE,
 		USB_DEVICE_ID_APPLE_WELLSPRINGT2_J680), .driver_data = 0 },
 	{ HID_USB_DEVICE(USB_VENDOR_ID_APPLE,
+		USB_DEVICE_ID_APPLE_WELLSPRINGT2_J680_ALT), .driver_data = 0 },
+	{ HID_USB_DEVICE(USB_VENDOR_ID_APPLE,
 		USB_DEVICE_ID_APPLE_WELLSPRINGT2_J213), .driver_data = 0 },
 	{ HID_USB_DEVICE(USB_VENDOR_ID_APPLE,
 		USB_DEVICE_ID_APPLE_WELLSPRINGT2_J214K), .driver_data = 0 },
